File: app/src/main/java/com/stedi/randomimagegenerator/app/model/repository/PresetRepository.java
package com.stedi.randomimagegenerator.app.model.repository;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.stedi.randomimagegenerator.app.model.data.Preset;

import java.util.List;

public interface PresetRepository {
    void save(@NonNull Preset preset) throws Exception;

    void remove(int id) throws Exception;

    @Nullable
    Preset get(int id) throws Exception;

    @NonNull
    List<Preset> getAll() throws Exception;
}

File: app/src/main/java/com/stedi/randomimagegenerator/app/model/repository/PresetRepository.kt
package com.stedi.randomimagegenerator.app.model.repository

import com.stedi.randomimagegenerator.app.model.data.Preset

interface PresetRepository {
    @Throws(Exception::class)
    fun save(preset: Preset)

    @Throws(Exception::class)
    fun remove(id: Int)

    @Throws(Exception::class)
    fun get(id: Int): Preset?

    @Throws(Exception::class)
    fun getAll(): List<Preset>
}